pkg/hubble/relay/pool/manager.go

```diff
@@ -236,20 +236,40 @@ func (m *PeerManager) List() []poolTypes.Peer {
 	return peers
 }
 
-func (m *PeerManager) add(hp *peerTypes.Peer) {
+func (m *PeerManager) put(hp *peerTypes.Peer) {
 	if hp == nil {
 		return
 	}
-	p := &peer{Peer: *hp}
 	m.mu.Lock()
-	m.peers[p.Name] = p
+
+	p := m.peers[hp.Name]
+
+	if p != nil && peerEqual(p.Peer, *hp) {
+		// Nothing changed, we don't need to reconnect
+		m.mu.Unlock()
+		return
+	}
+
+	if p != nil {
+		// Close old connection
+		m.disconnect(p)
+	}
+	m.peers[hp.Name] = &peer{Peer: *hp}
 	m.mu.Unlock()
 	select {
 	case <-m.stop:
 	case m.updated <- hp.Name:
 	}
 }
 
+func peerEqual(a, b peerTypes.Peer) bool {
```

Comment:
I'm a bit worried about future changes to the peer type invalidating this function. Would it make sense to move this into `peerTypes`, and maybe even make it a method of the `Peer` struct, such that it is closer to the definition of its fields?

Reply:
Good point, adding an `Equal` method makes more sense.